include/cocaine/api/connect.hpp

```diff
@@ -80,7 +80,7 @@ class client:
     typename traits<Event>::upstream_type
     invoke(const std::shared_ptr<typename traits<Event>::dispatch_type>& dispatch, Args&&... args) {
         if(!m_session) {
-            throw cocaine::error_t("client is not connected");
+            throw std::system_error(std::make_error_code(std::errc::not_connected));
```

Comment:
According to the http://en.cppreference.com/w/cpp/error/system_error/system_error a `what_arg` should contain an explanation string.

I think the `std::errc::not_connected` contains enough information to identify the error.